tests/parser.py

```diff
@@ -47,13 +47,15 @@ def read(file_name, test_filter):
                 if line != '\n':
                     test_lines.append(line)
                 else:
-                    test_struct = TestParser.__read_test_struct(test_lines, test_suite)
-                    if fnmatch("%s.%s" % (test_suite, test_struct.name), test_filter):
-                        tests.append(test_struct)
-                    test_lines = []
-            test_struct = TestParser.__read_test_struct(test_lines, test_suite)
-            if fnmatch("%s.%s" % (test_suite, test_struct.name), test_filter):
-                tests.append(test_struct)
+                    if test_lines:
+                        test_struct = TestParser.__read_test_struct(test_lines, test_suite)
```

Comment:
Can we hoist out the two bits of duplicated code into a lambda? Probably defined just after `test_suite` so the lambdas will bind to the variables correctly.

Reply:
This duplicated code is a bit ugly indeed, but AFAIK using a lambda here is not possible because Python doesn't support multi-line lambdas...

Reply:
Didn't know that about lambdas. What about a regular function instead?
```
>>> l = []
>>> def foo():
...     print(l)
...
>>> foo()
[]
>>> l.append(1)
>>> l.append('asdf')
>>> foo()
[1, 'asdf']
>>>
```